src/main/java/io/cryostat/core/net/JmxFlightRecorderService.java

```diff
@@ -0,0 +1,304 @@
+        delegate.enable();
+    }
+
+    @Override
+    public String getVersion() {
+        return delegate.getVersion();
+    }
+
+    @Override
+    public IDescribedMap<String> getDefaultRecordingOptions() {
+        return delegate.getDefaultRecordingOptions();
+    }
+
+    @Override
+    public IDescribedMap<EventOptionID> getDefaultEventOptions() {
+        return delegate.getDefaultEventOptions();
+    }
+
+    @Override
+    public IRecordingDescriptor start(
+            IConstrainedMap<String> recordingOptions,
+            String templateName,
+            TemplateType preferredTemplateType)
+            throws io.cryostat.core.FlightRecorderException, FlightRecorderException,
+                    ConnectionException, IOException, FlightRecorderException,
+                    ServiceNotAvailableException, QuantityConversionException, EventOptionException,
+                    EventTypeException {
+        return delegate.start(recordingOptions, enableEvents(templateName, preferredTemplateType));
+    }
+
+    private IConstrainedMap<EventOptionID> enableEvents(
+            String templateName, TemplateType templateType)
+            throws ConnectionException, IOException, io.cryostat.core.FlightRecorderException,
+                    FlightRecorderException, ServiceNotAvailableException,
+                    QuantityConversionException, EventOptionException, EventTypeException {
+        if (templateName.equals("ALL")) {
+            return enableAllEvents();
+        }
+        // if template type not specified, try to find a Custom template by that name. If none,
+        // fall back on finding a Target built-in template by the name. If not, throw an
+        // exception and bail out.
+        TemplateType type = getPreferredTemplateType(conn, templateName, templateType);
+        return conn.getTemplateService().getEvents(templateName, type).get();
+    }
+
+    private IConstrainedMap<EventOptionID> enableAllEvents()
+            throws ConnectionException, IOException, FlightRecorderException,
+                    ServiceNotAvailableException, QuantityConversionException, EventOptionException,
+                    EventTypeException {
+
+        IConnectionHandle handle = conn.getHandle();
+        EventOptionsBuilder builder =
+                new EventOptionsBuilder(
+                        cw, conn, () -> FlightRecorderServiceV2.isAvailable(handle));
+
+        for (IEventTypeInfo eventTypeInfo : conn.getService().getAvailableEventTypes()) {
+            builder.addEvent(eventTypeInfo.getEventTypeID().getFullKey(), "enabled", "true");
+        }
+
+        return builder.build();
+    }
+
+    private TemplateType getPreferredTemplateType(
+            JFRConnection connection, String templateName, TemplateType templateType)
+            throws io.cryostat.core.FlightRecorderException {
+        if (templateType != null) {
+            return templateType;
+        }
+        if (templateName.equals("ALL")) {
```

Comment:
Wondering if we still need this check since its only called in `enableEvents` and we already have:

```java
if (templateName.equals("ALL")) {
    return enableAllEvents();
}
```